src/Adapter/MSTest.CoreAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -357,37 +357,56 @@ private bool ProcessTestDataSourceTests(UnitTestElement test, MethodInfo methodI
             foreach (var dataSource in testDataSources)
             {
                 var data = dataSource.GetData(methodInfo);
-                var discoveredTests = new List<UnitTestElement>();
+                var discoveredTests = new Dictionary<string, UnitTestElement>();
+                var discoveredIndex = new Dictionary<string, int>();
                 var serializationFailed = false;
+                var index = 0;
 
                 foreach (var d in data)
                 {
                     var discoveredTest = test.Clone();
                     discoveredTest.DisplayName = dataSource.GetDisplayName(methodInfo, d);
 
+                    // if we have a duplicate test name don't expand the test, bail out.
+                    if (discoveredTests.ContainsKey(discoveredTest.DisplayName))
+                    {
+                        var firstSeen = discoveredIndex[discoveredTest.DisplayName];
+                        var warning = string.Format(CultureInfo.CurrentCulture, Resource.CannotExpandIDataSourceAttribute_DuplicateDisplayName, firstSeen, index, discoveredTest.DisplayName);
+                        warning = string.Format(CultureInfo.CurrentUICulture, Resource.CannotExpandIDataSourceAttribute, test.TestMethod.ManagedTypeName, test.TestMethod.ManagedMethodName, warning);
+                        PlatformServiceProvider.Instance.AdapterTraceLogger.LogWarning($"DynamicDataEnumarator: {warning}");
```

Comment:
It's weird to have the message translated but not this prefix part, I would like to have everything translated.

Reply:
Prefix was actually method name one, it's still something like that. (Consistent with other warnings in TestFx)

Reply:
Ok then let's ignore that but that's something we should probably address in the future, that's weird to mix technical (trace log) and functional to me.